### PR TITLE
[fontconfig] update to 2.13.94

### LINKED ISSUE #18172: [fontconfig] build failure on x64-osx
**Host Environment**
- OS: macOS 11
- Compiler: clang

**To Reproduce**
`./vcpkg install`

**Failure logs** 
vcpkg/buildtrees/fontconfig/build-x64-osx-dbg-err.log:
```
...
In file included from fcfreetype.c:48:
In file included from ./fcftint.h:26:
In file included from ../fontconfig/fcfreetype.h:27:
/usr/local/include/ft2build.h:37:10: fatal error: 'config/ftheader.h' file not found
#include "config/ftheader.h"
         ^~~~~~~~~~~~~~~~~~~
3 warnings and 1 error generated.
3 warnings generated.
make[3]: *** [fcfreetype.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from fcptrlist.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcptrlist.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcptrlist.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
3 warnings generated.
In file included from fcinit.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcmatch.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcmatch.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcmatch.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fchash.c:22:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fchash.c:22:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fchash.c:22:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcinit.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcinit.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcmatrix.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcmatrix.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fcmatrix.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
fchash.c:179:8: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_thread_fence() from <stdatomic.h> instead [-Wdeprecated-declarations]
         (b = fc_atomic_ptr_get (prev)); prev = &(b->next))
              ^
./fcatomic.h:78:32: note: expanded from macro 'fc_atomic_ptr_get'
#define fc_atomic_ptr_get(P)            (OSMemoryBarrier (), (void *) *(P))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:750:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
fchash.c:186:8: warning: 'OSAtomicCompareAndSwapPtrBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_compare_exchange_strong() from <stdatomic.h> instead [-Wdeprecated-declarations]
                if (!fc_atomic_ptr_cmpexch (prev, b, bucket))
                     ^
./fcatomic.h:80:38: note: expanded from macro 'fc_atomic_ptr_cmpexch'
#define fc_atomic_ptr_cmpexch(P,O,N)    OSAtomicCompareAndSwapPtrBarrier ((void *) (O), (void *) (N), (void **) (P))
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:548:6: note: 'OSAtomicCompareAndSwapPtrBarrier' has been explicitly marked deprecated here
bool    OSAtomicCompareAndSwapPtrBarrier( void *__oldValue, void *__newValue, void * volatile *__theValue );
        ^
fchash.c:196:10: warning: 'OSAtomicCompareAndSwapPtrBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_compare_exchange_strong() from <stdatomic.h> instead [-Wdeprecated-declarations]
    if (!fc_atomic_ptr_cmpexch (prev, b, bucket))
         ^
./fcatomic.h:80:38: note: expanded from macro 'fc_atomic_ptr_cmpexch'
#define fc_atomic_ptr_cmpexch(P,O,N)    OSAtomicCompareAndSwapPtrBarrier ((void *) (O), (void *) (N), (void **) (P))
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:548:6: note: 'OSAtomicCompareAndSwapPtrBarrier' has been explicitly marked deprecated here
bool    OSAtomicCompareAndSwapPtrBarrier( void *__oldValue, void *__newValue, void * volatile *__theValue );
        ^
fchash.c:228:13: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_thread_fence() from <stdatomic.h> instead [-Wdeprecated-declarations]
         (bucket = fc_atomic_ptr_get (prev)); prev = &(bucket->next))
                   ^
./fcatomic.h:78:32: note: expanded from macro 'fc_atomic_ptr_get'
#define fc_atomic_ptr_get(P)            (OSMemoryBarrier (), (void *) *(P))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:750:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
fchash.c:232:11: warning: 'OSAtomicCompareAndSwapPtrBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_compare_exchange_strong() from <stdatomic.h> instead [-Wdeprecated-declarations]
            if (!fc_atomic_ptr_cmpexch (prev, bucket, bucket->next))
                 ^
./fcatomic.h:80:38: note: expanded from macro 'fc_atomic_ptr_cmpexch'
#define fc_atomic_ptr_cmpexch(P,O,N)    OSAtomicCompareAndSwapPtrBarrier ((void *) (O), (void *) (N), (void **) (P))
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:548:6: note: 'OSAtomicCompareAndSwapPtrBarrier' has been explicitly marked deprecated here
bool    OSAtomicCompareAndSwapPtrBarrier( void *__oldValue, void *__newValue, void * volatile *__theValue );
        ^
3 warnings generated.
3 warnings generated.
8 warnings generated.
In file included from fclang.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fclang.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fclang.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fclang.c:26:
In file included from ./fcftint.h:26:
In file included from ../fontconfig/fcfreetype.h:27:
/usr/local/include/ft2build.h:37:10: fatal error: 'config/ftheader.h' file not found
#include "config/ftheader.h"
         ^~~~~~~~~~~~~~~~~~~
In file included from fclist.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:131:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefInc      (FcRef *r) { return fc_atomic_int_add (r->count, +1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fclist.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:132:56: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefDec      (FcRef *r) { return fc_atomic_int_add (r->count, -1); }
                                                       ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
In file included from fclist.c:25:
In file included from ./fcint.h:63:
./fcatomic.h:133:63: warning: 'OSAtomicAdd32Barrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_fetch_add() from <stdatomic.h> instead [-Wdeprecated-declarations]
static inline int    FcRefAdd      (FcRef *r, int v) { return fc_atomic_int_add (r->count, v); }
                                                              ^
./fcatomic.h:76:35: note: expanded from macro 'fc_atomic_int_add'
#define fc_atomic_int_add(AI, V)        (OSAtomicAdd32Barrier ((V), &(AI)) - (V))
                                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/OSAtomicDeprecated.h:162:9: note: 'OSAtomicAdd32Barrier' has been explicitly marked deprecated here
int32_t OSAtomicAdd32Barrier( int32_t __theAmount, volatile int32_t *__theValue );
        ^
3 warnings generated.
3 warnings generated.
3 warnings and 1 error generated.
make[3]: *** [fclang.lo] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

```

### PR DESCRIPTION
**Pull request by autoantwort**
Updates the port and fixes #18172

### CODE REVIEW

File: ports/fontconfig/portfile.cmake [hunk 1]
```diff
@@ -1,50 +1,21 @@
...
-    OPTIONS_DEBUG
-        "--with-libiconv-lib=${CURRENT_INSTALLED_DIR}/debug/lib"
-        ${OPT_DBG}
-    OPTIONS_RELEASE
-        "--with-libiconv-lib=${CURRENT_INSTALLED_DIR}/lib"
-        ${OPT_REL}
-    ADD_BIN_TO_PATH
-    ADDITIONAL_MSYS_PACKAGES xz findutils gettext gettext-devel  # for autopoint
-)
+vcpkg_configure_meson(SOURCE_PATH ${SOURCE_PATH})
```
Comment (JackBoosY) [inline ports/fontconfig/portfile.cmake:16]: Hmm... we prefer to use cmake, and the meson always has some problems.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/fontconfig/portfile.cmake (hunk 1 above)
Comment (autoantwort) [reply ports/fontconfig/portfile.cmake:16]: But fontconfig has no cmake support?

**Comment by JackBoosY** [comment]
> 2.13.94 is still only a release candidate for 2.14 and not an official release

Yes, for the tool ports, I suggest to use the official release version.

**Comment by ras0219** [comment]
2.13.1: Aug 30 2018
2.13.91: Jun 10 2019
2.13.92: Aug 9 2019
2.13.93: Nov 27 2020
2.13.94: Jun 28 2021

There hasn't been a stable release in almost 3 years but code is still being written that fixes real bugs. There is some point that we must move to consider the RC's stable enough, but I'd love to hear evidence from either side.

https://github.com/microsoft/vcpkg/issues/18172 is certainly evidence in support of moving to the RCs.

**Comment by dg0yt** [comment]
> There is some point at which we must consider the RCs stable enough, but I'd love to hear evidence from either side.

Some evidence from Repology:

[Packaging status](https://repology.org/project/fontconfig/versions)

**Comment by autoantwort** [comment]
@Neumann-A It currently fails to cross build from arm64-osx to x64-osx with some strange liker stuff, so it builds everything as x86_64 but then at the end want to link a arm64 binary. Maybe a bug in `vcpkg_install_meson`?
[package-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/6935205/package-x64-osx-dbg-out.log)
[meson-cross-x64-osx-debug.log](https://github.com/microsoft/vcpkg/files/6935207/meson-cross-x64-osx-debug.log)
[meson-cross-x64-osx-release.log](https://github.com/microsoft/vcpkg/files/6935208/meson-cross-x64-osx-release.log)
[meson-cross-x64-osx.log](https://github.com/microsoft/vcpkg/files/6935209/meson-cross-x64-osx.log)
[cmake-vars-x64-osx-dbg.cmake.log](https://github.com/microsoft/vcpkg/files/6935211/cmake-vars-x64-osx-dbg.cmake.log)

**Comment by autoantwort** [comment]
> @Neumann-A It currently fails to cross build from arm64-osx to x64-osx with some strange liker stuff, so it builds everything as x86_64 but then at the end want to link a arm64 binary. Maybe a bug in `vcpkg_install_meson`?

It seems that this only happens when you build the vcpkg tool as arm64 binary, when you build the tool as x64 binary via `arch -x86_64 ./bootstrap-vcpkg.sh` the build succeeds

**Review by ras0219-msft — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks @dg0yt for pointing to Repology, what's good enough for Fedora and MinGW should be good enough for us. :)

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**